lisa/sut_orchestrator/azure/notifiers.py

```diff
@@ -0,0 +1,180 @@
+
+from lisa import messages, notifier, schema
+from lisa.environment import EnvironmentMessage, EnvironmentStatus
+from lisa.util import LisaException, constants
+from lisa.util.logger import Logger
+
+from .. import AZURE
+
+
+@dataclass
+class VmsizeBasicInfo:
+    name: str = ""
+    vm_family: str = ""
+    core_count: int = 0
+
+
+@dataclass
+class EnvironmentVmsizeInfo:
+    name: str
+    status: EnvironmentStatus = EnvironmentStatus.New
+    vm_sizes: List[VmsizeBasicInfo] = field(default_factory=list)
+
+
+@dataclass_json()
+@dataclass
+class AzureNotifierSchema(schema.Notifier):
+    # fields to query test failure and test cases
+    file_name: str = "azure_usage_counter.json"
+
+
+class AzureNotifier(notifier.Notifier):
+    """
+    This notifier is used to collect the core usage information, which can then guide
+    the quota request in the subscription.
+    """
+
+    def __init__(self, runbook: AzureNotifierSchema) -> None:
+        notifier.Notifier.__init__(self, runbook)
+        self._file_name = runbook.file_name
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "azure_usage_counter"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return AzureNotifierSchema
+
+    def _received_message(self, message: messages.MessageBase) -> None:
+        if isinstance(message, EnvironmentMessage):
+            self._process_environment_message(message)
+        else:
+            raise LisaException(f"unsupported message received, {type(message)}")
+
+    def _subscribed_message_type(self) -> List[Type[messages.MessageBase]]:
+        return [EnvironmentMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        env_path = constants.RUN_LOCAL_LOG_PATH / "environments"
+        env_path.mkdir(exist_ok=True, parents=True)
+        self._file_path = env_path / f"{self._file_name}.json"
+        self._counter_log_path = env_path / "azure_usage_counter_collection.log"
+        self._locations_data_cache: Dict[str, Any] = {}
+        self._created_envs: Set[str] = set()
+        self._max_cores: Dict[str, int] = {}
+        self._current_cores: Dict[str, int] = {}
+        self._write_header = True
+
+    def _process_environment_message(self, msg: EnvironmentMessage) -> None:
+        if (
+            msg.status != EnvironmentStatus.Deployed
+            and msg.status != EnvironmentStatus.Deleted
+        ):
+            return
+
+        if (
+            msg.status == EnvironmentStatus.Deleted
+            and msg.name not in self._created_envs
+        ):
+            # Some skipped runs caused by no available environment has deleted message
+            # without deployed message. Ignore it.
+            return
+
+        env = EnvironmentVmsizeInfo(name=msg.name, status=msg.status)
+        assert msg.runbook.nodes_requirement
+        for node in msg.runbook.nodes_requirement:
+            from .common import AzureLocation, AzureNodeSchema
```

Comment:
It doesn't need to import inline, move them to the top of the file.

Reply:
Please check history of project.toml, the `requests` package should be included in Azure too.

Reply:
The above error is generated when "lisa --debug" test. There are just some dependency packages installed. There are no "requests" or "Azure" related packages. Even though I move `requests `package in `dependencies`,  there are no module named "Azure" errors.